lib/relationship_ancestry_converter_helper.rb

```diff
@@ -0,0 +1,113 @@
+module RelationshipAncestryConverterHelper
+  # src is relationship, dest is vm
+  def transfer_relationships_to_ancestry(dest_model, resource_type, relationship)
+    ancestry_resources = ancestry_resource_ids(relationship, resource_type, dest_model.rails_sequence_range(dest_model.my_region_number))
+    ancestry_sources = ancestry_src_ids(ancestry_resources)
+    new_ancestors = ancestry_of_src_ids_for_src(ancestry_sources)
+    update_src(new_ancestors, dest_model)
+  end
+
+  # src is vm, dest is relationship
+  def create_relationships_from_ancestry(src_model, dest_model, resource_type, relationship)
+    children = src_model.select("ancestry::bigint").where("ancestry not like '%/%'")
+    rels = src_model.where.not(:ancestry => nil).or(src_model.where(:id => children)).map do |obj|
+      dest_model.create!(:relationship  => relationship,
+                         :resource_type => resource_type,
+                         :resource_id   => obj.id,
+                         :ancestry      => obj.ancestry)
+    end.index_by(&:resource_id)
+    rels.each do |_, r|
+      next if r.ancestry.nil?
+
+      ancestry = r.ancestry.split('/').map { |rel| rels[rel.to_i].id }.join('/')
+      r.update!(:ancestry => ancestry)
+    end
+  end
+
+  private
+
+  #
+  # build sql that converts relationship ancestry with the id of the relationship to model-side ancestry with the id of the model
+  #
+
+  def ancestry_resource_ids(relationship, resource_type, id_range)
+    <<-SQL
+    SELECT a_rels.resource_id AS src_id, relationships.id AS rel_id, relationships.indx AS rel_indx
+    FROM   relationships a_rels
+    LEFT JOIN LATERAL UNNEST(STRING_TO_ARRAY(a_rels.ancestry, '/')::BIGINT[])
+    WITH ORDINALITY AS relationships(id, indx) ON TRUE
+    WHERE  a_rels.relationship = '#{relationship}'
+    AND    a_rels.resource_type = '#{resource_type}'
+    AND    a_rels.resource_id BETWEEN #{id_range.first} AND #{id_range.last}
+    SQL
+
+    # The above is a pure-SQL version of this similar Ruby code:
+    #
+    # def ancestry_resource_ids(relationship, resource_type, id_range)
```

Comment:
redundant method definition in the comment.  In other words, suggest to remove the `def ancestry_resource_ids(...)` in the comments and the correlating `end`

Reply:
I don't agree. I think the definitions are useful here because tracking the method params has been really important for catching bugs and understanding what this is doing.